service/common/src/main/java/org/apache/polaris/service/auth/PemUtils.java

```diff
@@ -46,9 +45,38 @@ private static byte[] parsePEMFile(Path pemPath) throws IOException {
       throw new FileNotFoundException(
           String.format("The file '%s' doesn't exist.", pemPath.toAbsolutePath()));
     }
-    try (PemReader reader = new PemReader(Files.newBufferedReader(pemPath, UTF_8))) {
-      PemObject pemObject = reader.readPemObject();
```

Comment:
One general remark: the original `readPemObject()` method, in principle, only reads the first object in the PEM file:

https://downloads.bouncycastle.org/java/docs/bcprov-jdk18on-javadoc/org/bouncycastle/util/io/pem/PemReader.html#readPemObject()

With your changes, what happens when the PEM file contains more than one object? And what if it doesn't contain any? It might be good to add tests for these cases as well.

Reply:
Thanks for the feedback and noting the behavior of `readPemObject()`. I adjusted the new implementation to stop reading when finding a footer line, and added unit tests for that scenario, as well as an empty file, which throws an `IOException`.